### DISCUSSION TITLE
[Get-MonkeyMSALToken] - Acquire token failed. An error occurred while sending the request.

### DISCUSSION THREAD

**Discussion by toffitomek** [silverhack/monkey365#111] [General]
I was trying to run Monkey 365 different ways but I am always getting following error:
`WARNING: [12:49:21:812] - [Get-MonkeyMSALToken] - Acquire token failed. An error occurred while sending the request. - warning - servername.domain.com - `

I don't even get login prompt.

Any ideas?

I am running PS 5.1 on windows server 2016

**Comment by silverhack** [comment]
Hi, 

Without more information is hard to say. Most common errors are due to conditional access issues.

I'd recommend to add verbose, debug and informationAction flags to monkey365 to see what's happen. 

Hope that helps

**Comment by toffitomek** [comment]
Thanks for answer, 

Conditional access should be ok, plus it's reviewed after 'login' and here I don't even get a prompt. Never had issues running other tools and it's trusted location. Also e.g. CIS Scuba works fine.

sadly extra flags doesn't help a lot, -Debug just prints original error doesn't give more info, see below:

<img width="1488" alt="image" src="https://github.com/user-attachments/assets/1c867cfd-16d2-4fab-82bf-2d557541556b">

<img width="1598" alt="image" src="https://github.com/user-attachments/assets/66481b3d-ce7c-432e-ab1a-31302e74284f">

**Comment by silverhack** [comment]
Hi, 

Thanks for the detailed response. Which parameters are you using? It's a gov tenant? 

As far as I can see the issue is with the internal HTTP client, so in parallel I will try to test the tool in a 2k16 server and I'll let you know. 

Thanks

**Comment by silverhack** [comment]
Hi again, 

Tested in various servers and works as expected with no errors. On the other hand, I'm receiving the same error as you with Windows Server 2016. 2K16 tries to negotiate over TLS 1.X which is not supported by Microsoft 365. You should apply a patch in your OS to force a TLS 1.2 connection, and potentially update the OS and .NET framework with the latest patches. 

In case you can't (you're not an administrator), you can use the following workaround:

1. Open PowerShell 
2. Execute the following **[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12**
3. Run Monkey365

Hope that helps.
Cheers,

**Reply by toffitomek** [reply]
Thanks a lot, this seems to do the trick!

case closed
